### PR TITLE
Add shared database docs

### PR DESCRIPTION
**Pull request by rmweir**
Preview platform: https://deploy-preview-314--vcluster-docs-site.netlify.app/docs/platform/administer/connector/database
Preview vcluster: https://deploy-preview-314--vcluster-docs-site.netlify.app/docs/vcluster/next/configure/vcluster-yaml/control-plane/components/backing-store/database/external

### CODE REVIEW

File: platform/administer/shared-database/shared-database.mdx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
+title: Shared database vCluster Platform
+sidebar_label: Shared Database
+sidebar_position: 1
```
Comment (Piotr1215) [inline platform/administer/shared-database/shared-database.mdx:4]: Could you also add `description` to the front matter. This helps discover the docs by search engines.

```suggestion
sidebar_position: 1
description: Learn how to use shared database with vcluster.
```

File: platform/administer/shared-database/shared-database.mdx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+---
+
+The shared database feature allows an admin to configure a database server to be used to manage backing stores for multiple vCluster. For each vCluster, the feature automatically creates individual databases and
```
Comment (Piotr1215) [inline platform/administer/shared-database/shared-database.mdx:7]:
```suggestion
The shared database feature allows an admin to configure a database server to be used to manage backing stores for multiple virtual clusters. For each virtual cluster, the feature automatically creates individual databases and
```

File: platform/administer/shared-database/shared-database.mdx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+* Contain the `loft.sh/connector-type` label with the `shared-database` value.
+* Reside in the same namespace as vCluster Platform.
+* Contain the fields:
+    * endpoint
+    * password
+    * port
+    * user
+
+The template for a shared database connector secret is:
+```
```
Comment (Piotr1215) [inline platform/administer/shared-database/shared-database.mdx:38]: We would like to have titles on the code blocks, maybe something like:
\```yaml title="Shared database secret"

File: platform/administer/shared-database/shared-database.mdx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  endpoint: <endpoint> # this could be a service endoint or an external hostname endpoint
+  password: <password>
+  port: <port> # 3306 is a common port for database servers
+  user: <user>
+
+```
+
+1. Replace all values surrounded by angular brackets with your own value and save to a file as `<filename>`.
+2. Create the secret:
+`kubectl create -f <filename>`
```
Comment (Piotr1215) [inline platform/administer/shared-database/shared-database.mdx:56]:
```suggestion
2. Create the secret: `kubectl create -f <filename>`
```

File: platform/administer/shared-database/shared-database.mdx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+That's it. A new vCluster can now reference the created connector secret by setting the `controlPlane.backingStore.database.external.connector` field in the vCluster config
+to the `name` of the connector secret.
+
+#### Example
+
+The following example is a connector secret for the shared database feature. It is configured for an internal instance of mysql that is exposed
+with a ClusterIP service named `mysql` in the `vcluster-platform` namespace (let us presume this is the namespace vCluster Platform is installed in). This could
+instead be any other endpoint, such as an AWS RDS MySQL endpoint.
+
+```
```
Comment (Piotr1215) [inline platform/administer/shared-database/shared-database.mdx:67]: Similar here, maybe:
\```yaml title="Endpoint secret"

**Review by Piotr1215 — COMMENTED** [review]
Thank you for submitting the doc @rmweir , added a few comments/suggestions.

File: platform/administer/shared-database/shared-database.mdx (hunk 1 above)
Comment (rmweir) [reply platform/administer/shared-database/shared-database.mdx:4]: Done, I took the description you provided :P.

**Comment by rmweir** [comment]
@Piotr1215 addressed feedback and added a [warning](https://github.com/loft-sh/vcluster-docs/pull/314/files#diff-6885da81aabd0fa338bc2d6fc9553d177cb9649272e49de9bb79d8a6b02c55e2R61) regarding the database secrets.

File: platform/administer/shared-database/shared-database.mdx (hunk 5 above)
Comment (rmweir) [reply platform/administer/shared-database/shared-database.mdx]: Used different title but done.

File: platform/administer/shared-database/shared-database.mdx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  password: <password>
+  port: <port> # 3306 is a common port for database servers
+  user: <user>
+
+```
+
+1. Replace all values surrounded by angular brackets with your own value and save to a file as `<filename>`.
+2. Create the secret: `kubectl create -f <filename>`
+
+That's it. A new vCluster can now reference the created connector secret by setting the `controlPlane.backingStore.database.external.connector` field in the vCluster config
```
Comment (sowmyav27) [inline platform/administer/shared-database/shared-database.mdx:58]: Can we add a vCluster.yaml here instead so it will be easier for users to configure ?

**Review by sowmyav27 — CHANGES_REQUESTED** [review]

File: platform/administer/shared-database/shared-database.mdx (hunk 6 above)
Comment (rmweir) [reply platform/administer/shared-database/shared-database.mdx]: Added a full separate doc for vCuster now.

**Review by Piotr1215 — COMMENTED** [review]
Thank you for writing the doc @rmweir. Looks like there are some unresolved links issues in the build. Could you please [allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so I can help you with it?

**Review by Piotr1215 — COMMENTED** [review]
@rmweir could you please check if my style and writing modifications didn't mess up with the content?

File: platform/administer/connector/database.mdx [hunk 7]
```diff
@@ -2,37 +2,42 @@
...
 ---
 
-The database connector feature allows an admin to configure a database server to be used to manage backing stores for multiple virtual clusters. For each virtual cluster, the feature automatically creates individual databases and
-creates a non-privileged user. Read more about vCluster's using of external
-databases in the [vCluster External Database docs](/docs/vcluster/configure/vcluster-yaml/control-plane/components/backing-store/database/external).
+import BasePrerequisites from '../../_partials/install/base-prerequisites.mdx';
 
+import InstallNextSteps from '../../_partials/install/install-next-steps.mdx';
+
+The database connector feature allows an admin to configure a database server to manage backing stores for multiple virtual clusters. For each virtual cluster, the feature automatically creates individual databases and non-privileged users. Read more about the use of external databases in the [External Database documentation](/docs/vcluster/configure/vcluster-yaml/control-plane/components/backing-store/database/external).
```
Comment (rmweir) [inline platform/administer/connector/database.mdx:12]: `the feature automatically creates individual databases and non-privileged users` should be `the feature automatically creates an independent database and non-privileged user`. Each virtual cluster only gets one of these.

**Review by Piotr1215 — APPROVED** [review]

**Pull request merged**